File: m6/getQR.py
import ast
from PIL import Image, ImageColor
from  qrtools import QR

def getPoints(filename):
    with open(filename, 'r') as f:
        content = f.readlines()

    content = [x.strip() for x in content]

    points=[]
    MAX_X = -1
    MAX_Y = -1;

    for c in content:
        c = ast.literal_eval(c)
        MAX_X = max(MAX_Y, c[0])
        MAX_Y = max(MAX_Y, c[1])
        points.append(c)

    return points, MAX_X, MAX_Y

def createQRImage(filename, points):
    im = Image.new('1', (MAX_X + 1, MAX_Y + 1), "white")
    for p in points:
        im.putpixel((p[0], p[1]), ImageColor.getcolor('black', '1'))
    im.save(filename)
    return im

print "[+] Reading coordinates from file..."
points, MAX_X, MAX_Y = getPoints("code.txt")

print"[+] Creating QR code image..."
im = createQRImage('M6-QR.png', points)
print "\t'M6-QR.png' created.";

File: m6/getQR.cpp
/*
Needs: OpenCV installed
Compile: g++ -std=c++11 sol.cpp -o sol `pkg-config opencv --cflags --libs`
*/

#include <bits/stdc++.h>

#include <opencv2/core/core.hpp>
#include <opencv2/highgui/highgui.hpp>

using namespace cv;
using namespace std;

std::string extract_ints(std::ctype_base::mask category, std::string str, std::ctype<char> const& facet)
{
    using std::strlen;

    char const *begin = &str.front(),
               *end   = &str.back();

    auto res = facet.scan_is(category, begin, end);

    begin = &res[0];
    end   = &res[strlen(res)];

    return std::string(begin, end);
}

std::string extract_ints(std::string str)
{
    return extract_ints(std::ctype_base::digit, str,
         std::use_facet<std::ctype<char>>(std::locale("")));
}


int  main(int argc, char const *argv[]) {
  vector<pair<int, int> > points;
  int maxX = -1, maxY = -1;

  cout <<"[+] Reading coordinates from file...\n";
  ifstream codeFile("code.txt", ios::in);

  if(codeFile) {
    string line;
    while(getline(codeFile, line)) {
      istringstream iss(line);
      int x, y;
      char ignore;
      iss >> ignore >> x >> ignore >>  y >> ignore ;
      maxX = max(maxX, x);
      maxY = max(maxY, y);
      pair<int, int> p = make_pair(x, y);
      points.push_back(p);
    }
    codeFile.close();
  }

  cout <<"[+] Creating QR code image...\n";
  Mat img(maxX+1, maxY+1, CV_8UC1, Scalar(255));
  cout <<"\tImage width: " << img.cols <<"\n";
  cout <<"\tImage height: " << img.rows <<"\n";

  vector<pair<int, int> >::iterator it;
  for (it = points.begin() ; it != points.end() ; ++it) {
    int x = it->first;
    int y = it->second;
    img.at<uchar>(y, x) = 0x00;
  }

  imwrite("qrcode.png", img);
  cout <<"\t'qrcode.png' created.\n";
  cout <<"[++END++]\n";
  return 0;
}
